src/jackdaw/test/journal.clj

```diff
@@ -76,21 +78,28 @@
     (throw (ex-info "no message stream to journal:" machine)))
 
   (let [{:keys [messages]} (:consumer machine)]
-    (async/go-loop [record (async/<! messages)]
-      (when record
-        (journal-result machine record)
-        (recur (async/<! messages))))))
+    (d/loop [record (s/take! messages)]
+      (d/chain record
```

Comment:
@cddr Here and further... You can use `d/chain'` instead of `d/chain` any time you're sure that the argument would be a manifold deferred to avoid unnecessary runtime coercion.